coalib/parsing/CliParsing.py

```diff
@@ -87,3 +87,31 @@ def parse_custom_settings(sections,
                                origin=origin,
                                section_name=key_touple[0],
                                from_cli=True)
+
+
+def check_conflicts(cli_sections, log_printer):
+    '''
+    Checks if there are any conflicting aruments passed
+
+    :return:            True if no conflicts
+    :raises SystemExit: If there are conflicting arguments (exit code: 2)
+    '''
+    conflicts = defaultdict(set)
+
+    # Add any conflict definitions here
+    conflicts["no_config"].add("save")
+    conflicts["no_config"].add("find_config")
+
+    args = list(conflicts.keys())
+    for arg in args:
+        for conflict in conflicts[arg]:
+            conflicts[conflict].add(arg)
+
+    for section in cli_sections:
+        for arg in cli_sections[section]:
+            for conflict in conflicts[arg]:
```

Comment:
(from our gitter conevrsation)

& in sets means intersection:

```
In [4]: set([1,2,3]) & set([2,3,4])
Out[4]: {2, 3}
```

when you do:
```
for a in list1:
    if a in list2:
        # do something
```
it basically means you want the intersection of 2 lists
In such cases, it makes sense to use 2 sets (assuming you dont care about the ordering and duplicated)

Because `for + in` is O(n^2) while `set1 & set2` is O(2*n)

Reference: https://www.ics.uci.edu/~pattis/ICS-33/lectures/complexitypython.txt